tobias/utils/motifs.py

```diff
@@ -153,10 +154,15 @@ def from_file(self, path):
 					bg_and_freq = re.split(r"(?<=\d)\s+", line.strip()) # split after every number followed by a whitespace
 					bg_dict = {key: float(value) for key, value in [el.split(" ") for el in bg_and_freq]}
 					bg = np.array([bg_dict[base] for base in self[-1].bases])
-					self[-1].bg = bg	#list of 4 if ACGT
 
+				### Motif content ###
 				# parse id, name
 				elif line.startswith("MOTIF"):
+
+					#Initialize motif
+					probability_matrix = []
+					self.append(OneMotif(motifid="", counts=[[],[],[],[]]))	#initialize dummy OneMotif for filling in

```

Comment:
This solves the issue of the last entry being empty - the new motif is only added when there is a header line for a new motif ("MOTIF (...)").

I would like to initialize it first because it makes it easier to add the information as it is found - otherwise, you have to deal with assignment at the very end (after reading each motif), which makes it prone to forgetting previously defined variables. What is the issue in setting `counts = None` as default?

Reply:
Setting `counts = None` in itself is no problem. I was thinking about all the other functions in OneMotif that will not work if for some reason an empty object is called with them. Probably it is enough to think that someone who creates an empty object will then have to worry about not calling them. Anyway, I will change `counts = None` to default.